### PR TITLE
Restore xiaomi dev temp the correct way

### PR DESCRIPTION
**Pull request by rforro**
I think that rather than disabling the temperature sensor inside of various xiaomi sensors, we should leave this to user. So I decided to restore it, but not as environment temperature sensor. I've chosen for this device temperature cluster (what it actually is), so it should be more or less zigbee standard compliant.

Currently I implemented this for only following 2 devices, which I won, Aqara Water Leak and Aqara Window Sensor. I could extend it to all devices from #92 but I'd like to here your feedback first.

P.S. If this will be merged I'll do the necessary changes to `zha/sensor.py` and `zha/core/registries.py`

#834

### CODE REVIEW

File: zhaquirks/xiaomi/__init__.py [hunk 1]
```diff
@@ -414,6 +419,16 @@ class MotionCluster(LocalDataCluster, MotionOnEvent):
     reset_s: int = 70
 
 
+class DeviceTemperatureCluster(LocalDataCluster, DeviceTemperature):
+    """Device Temperature Cluster."""
+
+    cluster_id = DeviceTemperature.cluster_id
+
+    def update_attribute(self, attrid, value):
+        """attribute value reported"""
```
Comment (MattWestb) [inline zhaquirks/xiaomi/__init__.py:428]: """Attribute value reported."""

Comment (rforro) [reply zhaquirks/xiaomi/__init__.py:428]: That's not the main problem why is pipeline failing

Comment (MattWestb) [reply zhaquirks/xiaomi/__init__.py]: The py test is failing in other files that you have **not** making changes in and that test its broken or some other have doing bad commitments.

"Unchanged files with check annotations **Beta**"

**Review by MattWestb — APPROVED** [review]

**Comment by rforro** [comment]
The pipeline is OK, but don't merge it yet. The thing is, if the device IS publishing it's device temperature, but is missing this DeviceTemperature cluster, by calling `update_attribute` it will produce an error. To fix it I need to know which Aqara devices are publishing its internal temperature or somehow check if DeviceTemperature  was implemented.

**Comment by MattWestb** [comment]
The test have doing some "magic" :-))

For old devices you can looking in the old commits witch devices that have being deleting the temperature measurement cluster but for new i think its easiest looking if Z2M have it implanted in the devices = then is normally is working.

**Comment by rforro** [comment]
I've added this cluster to all devices from #92

File: zhaquirks/xiaomi/__init__.py [hunk 2]
```diff
@@ -414,6 +419,16 @@ class MotionCluster(LocalDataCluster, MotionOnEvent):
     reset_s: int = 70
 
 
+class DeviceTemperatureCluster(LocalDataCluster, DeviceTemperature):
+    """Device Temperature Cluster."""
+
+    cluster_id = DeviceTemperature.cluster_id
+
+    def update_attribute(self, attrid, value):
```
Comment (Adminiuga) [inline zhaquirks/xiaomi/__init__.py:427]: This should already be in the latest zigpy, so no need to redefine it here.

Comment (rforro) [reply zhaquirks/xiaomi/__init__.py:427]: Without `def update_attribute(self, attrid, value):` it wasn't working on my "testbench" HA 2012.12.07

Comment (rforro) [reply zhaquirks/xiaomi/__init__.py]: You were right, sorry

**Review by Adminiuga — APPROVED** [review]
LGTM

**Comment by Adminiuga** [comment]
This still requires changes to ZHA to expose this as sensors.

**Comment by rforro** [comment]
> This still requires changes to ZHA to expose this as sensors.

Yes, I've mentioned it, but I wasn't sure if this PR will be accepted.
> P.S. If this will be merged I'll do the necessary changes to `zha/sensor.py` and `zha/core/registries.py`

**Pull request merged**